packages/dev/core/src/Buffers/bufferUtils.ts

```
@@ -243,6 +271,96 @@ export function GetFloatData(
+ * @param forceCopy defines a boolean indicating that the returned array must be cloned upon returning it
+ * @returns a typed array containing vertex data
+ */
+export function GetTypedArrayData(
+    data: DataArray,
+    size: number,
+    type: number,
+    byteOffset: number,
+    byteStride: number,
+    normalized: boolean,
+    totalVertices: number,
+    forceCopy?: boolean
+): VertexDataTypedArray {
+    const typeByteLength = GetTypeByteLength(type);
+    const constructor = GetTypedArrayConstructor(type);
+    const count = totalVertices * size;
+
+    // Handle number[]
+    if (Array.isArray(data)) {
+        const offset = byteOffset / typeByteLength;
+        const stride = byteStride / typeByteLength;
+        const lastIndex = offset + (totalVertices - 1) * stride + size;
+
+        if (lastIndex > data.length) {
+            throw new Error("Last accessed index is out of bounds.");
+        }
+
+        if (stride !== size) {
+            const copy = new constructor(count);
+            EnumerateFloatValues(data, offset, stride, size, type, count, normalized, (values, index) => {
+                for (let i = 0; i < size; i++) {
+                    copy[index + i] = values[i];
+                }
+            });
+            return copy;
+        }
+
+        return new constructor(data);
+    }
+
+    // Handle ArrayBuffer and ArrayBufferView
+    let buffer: ArrayBuffer;
+    let adjustedByteOffset = byteOffset;
+
+    if (data instanceof ArrayBuffer) {
+        buffer = data;
+    } else {
+        buffer = data.buffer;
+        adjustedByteOffset += data.byteOffset;
+    }
+
+    const lastByteOffset = adjustedByteOffset + (totalVertices - 1) * byteStride + size * typeByteLength;
+    if (lastByteOffset > buffer.byteLength) {
+        throw new Error("Last accessed byte is out of bounds.");
+    }
+
+    const tightlyPackedByteStride = size * typeByteLength;
+    if (byteStride !== tightlyPackedByteStride) {

```

Should we throw an exception if `byteStride` is less than `tightlyPackedByteStride`?

---

Good idea.

I don't think it should ever happen, but then by that logic, I suppose the out-of-range should also never happen either... lol